target/linux/ramips/dts/mt7621_plasmacloud_pax1800-lite.dts

```diff
@@ -0,0 +1,2219 @@
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	compatible = "plasmacloud,pax1800-lite", "mediatek,mt7621-soc";
+	model = "Plasma Cloud PAX1800-Lite";
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	aliases {
+		led-boot = &led_status_green;
+		led-failsafe = &led_status_green;
+		led-running = &led_status_green;
+		led-upgrade = &led_status_green;
+		label-mac-device = &gmac1;
+	};
+
+	gpio_export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		dc_in {
+			gpio-export,name = "dc_in";
+			gpios = <&gpio 28 GPIO_ACTIVE_LOW>;
+		};
+
+		poe_in {
+			gpio-export,name = "poe_in";
+			gpios = <&gpio 29 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		status_blue {
+			color = <LED_COLOR_ID_BLUE>;
+			function = LED_FUNCTION_STATUS;
+			gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+		};
+
+		led_status_green: status_green {
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_STATUS;
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		status_red {
+			color = <LED_COLOR_ID_RED>;
+			function = LED_FUNCTION_STATUS;
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	watchdog {
+		compatible = "linux,wdt-gpio";
+		gpios = <&gpio 7 GPIO_ACTIVE_LOW>;
+		hw_algo = "toggle";
+		/* hw_margin_ms is actually 300s but driver limits it to 60s */
+		hw_margin_ms = <60000>;
+		always-running;
+	};
+};
+
+&i2c {
+	status = "okay";
+
+	g751: g751@48 {
+		compatible = "gmt,g751";
+		reg = <0x48>;
+		status = "okay";
+	};
+};
+
+&spi0 {
+	status = "okay";
+	num-cs = <2>;
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		spi-max-frequency = <20000000>;
+		reg = <0>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "uboot";
+				reg = <0x00000 0x80000>;
+			};
+
+			partition@80000 {
+				label = "uboot_env";
+				reg = <0x80000 0x10000>;
+			};
+
+			factory: partition@90000 {
+				label = "factory";
+				reg = <0x90000 0x80000>;
+			};
+
+			partition@110000 {
+				label = "keys";
+				reg = <0x110000 0x10000>;
+			};
+		};
+	};
+
+	flash@1 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "spi-nand";
+		spi-max-frequency = <20000000>;
+		reg = <1>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "firmware1";
+				reg = <0x0 0x4000000>;
+				compatible = "linux,ubi";
+			};
+
+			partition@4000000 {
+				label = "firmware2";
+				reg = <0x4000000 0x4000000>;
+				compatible = "linux,ubi";
+			};
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie1 {
+	wifi0: wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+
+		mediatek,mtd-eeprom = <&factory 0x0>;
```

Comment:
Use NVMEM.

Reply:
Thanks, updated. Please check the change.